addons/mail/models/mail_message.py

```diff
@@ -497,6 +497,13 @@ def _get_forbidden_access(self, operation: str) -> api.Self:
 
         for model, docid_msgids in model_docid_msgids.items():
             documents = self.env[model].browse(docid_msgids)
+            # Handle discuss channel read only
+            if model == "discuss.channel" and operation in ["write", "create"]:
+                for document in documents:
+                    # sudo: we need to check this status, even if the user
+                    # cannot read the thread.
+                    if document.sudo().read_only:
+                        forbidden += self.browse(messages_to_check)
```

Comment:
Probably to move in _get_mail_message_access override with proper ACLs this should be fine

Reply:
More details: we should probably link "posting a message" to "writing on readonly channel": if you are an admin allowed to write on a channel, technically you can post on it, so overriding ``_get_mail_message_access`` seems sufficient and aligned with current rules.

Reply:
> if you are an admin allowed to write on a channel

Nobody said that was true. Why would an admin be able to post on a **read-only** channel? If the channel is read-only but only for a subset of users, then it's just fancy permissions…
If we want fancy permissions, we should build that, but it's not what I'm currently building.

Reply:
Well I suppose some people have the right to make it not readonly anymore. Otherwise it is a dead end.

Reply:
Reading the diff, I see there is no protection against write on readonly channel. Which mean anyone that can write on channel can open it again, post, and close it. Hence it is not really security, so I don't see why you are trying to make things serious.

Reply:
> Well I suppose some people have the right to make it not readonly anymore. Otherwise it is a dead end.

You can uncheck the box making it writable again.

> Which mean anyone that can write on channel can open it again, post, and close it.

Yes, that's the point, it's read-only or not. 

> Hence it is not really security, so I don't see why you are trying to make things serious.

It's not security, it never was, it's a feature that makes a channel read-only :shrug:

Reply:
Then it's just fancy permissions :shrug:

Reply:
The read-only feature makes functionally more sense with owners/admins that can write on these read-only channels. There's an ongoing PR for adding owners/admins: https://github.com/odoo/odoo/pull/177445

Merging this PR and letting only admin write on read-only channels still has a few value, albeit I agree quite niche use. I wouldn't mind to wait owner/admins PR before merging this PR or merge this PR before the owner/admins PR, the end result is the same and feature has still some value in all cases.

Reply:
Admin can now post on read-only channels